warden/x/async/keeper/keeper.go

```diff
@@ -289,3 +289,48 @@ func (k Keeper) callEVMWithData(
 
 	return res, nil
 }
+
+type completedFuture struct {
+	future types.Future
+	result types.FutureResult
+}
+
+func (k Keeper) getCompletedFuturesWithoutValidatorVote(ctx context.Context, valAddress []byte, limit int) ([]completedFuture, error) {
+	it, err := k.futures.results.IterateRaw(ctx, nil, nil, collections.OrderAscending)
```

Comment:
i think it's better to go in reverse order so that new futures are handled first, otherwise after the first `limit` results you will never reach the new ones

```suggestion
	it, err := k.futures.results.IterateRaw(ctx, nil, nil, collections.OrderDescending)
```

Reply:
I mean, eventually we would process all the results anyway, wouldn't we?

Reply:
I actually think it would be better to use ascending order here. If new futures are processed first and added rapidly, we might never process some of the older ones.

Reply:
Hm, honestly I thinks ascending orders is makes more sense here. There is risk to never achieve the oldest ones if we will go from newest

Reply:
okay let's keep it ascending for no was there is not right answer, it really depends on the type of loads we'll see